helix-term/src/ui/document.rs

```diff
@@ -405,14 +408,16 @@ impl<'a> TextRenderer<'a> {
         }
 
         let width = grapheme.width();
+        let space = if is_virtual { " " } else { &self.space };
+        let nbsp = if is_virtual { " " } else { &self.nbsp };
         let grapheme = match grapheme {
             Grapheme::Tab { width } => {
                 let grapheme_tab_width = char_to_byte_idx(&self.tab, width);
```

Comment:
To properly render tabs you need to add a second field `virtual_tab` to which gets filled with white spaces in `new` instead of using whitespace chars.

Reply:
Gotcha. I believe its a bit verbose inside the Grapheme::Tab, but it should now handle the tab rendering

Reply:
Looks good now but I would prefer if you could move the if condition outside the match and handle it just like the other variables (so `let tab = ..`).

I thought it was quite clean how you did it for the other whitespace chars so it would be great if tab was the same

Reply:
That makes sense, done.